access-common/src/main/java/edu/unc/lib/dl/ui/service/FedoraContentService.java

```diff
@@ -15,235 +15,98 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 
+import edu.unc.lib.dl.acl.service.AccessControlService;
 import edu.unc.lib.dl.acl.util.AccessGroupSet;
-import edu.unc.lib.dl.acl.util.GroupsThreadStore;
-import edu.unc.lib.dl.fedora.AccessClient;
-import edu.unc.lib.dl.fedora.AuthorizationException;
-import edu.unc.lib.dl.fedora.FedoraException;
-import edu.unc.lib.dl.search.solr.model.BriefObjectMetadataBean;
-import edu.unc.lib.dl.search.solr.model.Datastream;
-import edu.unc.lib.dl.search.solr.model.SimpleIdRequest;
-import edu.unc.lib.dl.search.solr.util.SearchFieldKeys;
-import edu.unc.lib.dl.search.solr.util.SearchSettings;
-import edu.unc.lib.dl.ui.exception.InvalidRecordRequestException;
-import edu.unc.lib.dl.ui.exception.ResourceNotFoundException;
-import edu.unc.lib.dl.ui.util.AnalyticsTrackerUtil;
-import edu.unc.lib.dl.ui.util.AnalyticsTrackerUtil.AnalyticsUserData;
-import edu.unc.lib.dl.util.ContentModelHelper;
-import edu.unc.lib.dl.util.ContentModelHelper.DatastreamCategory;
+import edu.unc.lib.dl.fcrepo4.BinaryObject;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fedora.PID;
 
 /**
- * Connects to and streams datastreams from Fedora.
+ * Streams binary content from repository objects.
  *
  * @author bbpennel
  */
 public class FedoraContentService {
     private static final Logger LOG = LoggerFactory.getLogger(FedoraContentService.class);
 
-    private AccessClient accessClient;
-
-    private String fedoraHost;
-
-    @Autowired
-    private SearchSettings searchSettings;
-    @Autowired
-    protected SolrQueryLayerService queryLayer;
-
-    @Autowired(required = false)
-    protected AnalyticsTrackerUtil analyticsTracker;
-
-    private final int numberOfRetries = 1;
+    public static final String CONTENT_DISPOSITION = "Content-Disposition";
+
+    private AccessControlService accessControlService;
+
+    private RepositoryObjectLoader repositoryObjectLoader;
+
+    /**
+     * Set content headers and stream the binary content of the specified
+     * datastream from the object identified by pid.
+     *
+     * @param pid pid of object containing datastream
+     * @param dsName name of datastream being requested. If null, then original
+     *            file assume.
+     * @param principals principals of requesting client
+     * @param asAttachment if true, then content-disposition header will specify
+     *            as "attachment" instead of "inline"
+     * @param response response content and headers will be added to.
+     * @throws IOException
+     */
+    public void streamData(PID pid, String dsName, AccessGroupSet principals, boolean asAttachment,
+            HttpServletResponse response) throws IOException {
+        // Default datastream is DATA_FILE
+        String datastream = dsName == null ? ORIGINAL_FILE : dsName;
```

Comment:
i know this pattern is popular in the Fedora codebase, and i know it's not that hard to understand, but i tend to dislike the ternary operator + assignment in a single statement. it's less clear than, say

`String datastream;
if (dsName == null) {
    datastream = ORIGINAL_FILE;
} else {
    datastream = dsName;`
}
or even
` String datastream = dsName;
if (datastream == null) {
    datastream = ORIGINAL_FILE;
}`

Reply:
I feel like this is the situation where ternary operator are appropriate, since it is a simple if/else assignment. That is the impression I am getting from stackoverflow responses as well https://softwareengineering.stackexchange.com/questions/28314/ternary-operator-considered-harmful/28394

Reply:
hmm, the stackoverflow threads discussing this topic seem to have been closed due to being not constructive :-) i'm ok with it, but would really prefer parentheses at the very least for improved readability, if you wouldn't be totally opposed to that.